readthedocs/filetreediff/__init__.py

```diff
@@ -28,7 +32,120 @@
 MANIFEST_FILE_NAME = "manifest.json"
 
 
-def get_diff(version_a: Version, version_b: Version) -> FileTreeDiff | None:
+class FileTreeDiffFileStatus(StrEnum):
+    """Status of a file in the file tree diff."""
+
+    ADDED = auto()
+    DELETED = auto()
+    MODIFIED = auto()
+
+    @property
+    def emoji(self):
+        """Return an emoji representing the file status."""
+        if self == FileTreeDiffFileStatus.ADDED:
+            return "➕"
+        elif self == FileTreeDiffFileStatus.DELETED:
+            return "❌"
+        elif self == FileTreeDiffFileStatus.MODIFIED:
+            return "📝"
+        return ""
+
+
+class FileTreeDiffFile:
+    def __init__(
+        self,
+        filename: str,
+        current_version,
+        current_version_build: Build,
+        base_version,
+        base_version_build: Build,
+        status: FileTreeDiffFileStatus,
+        resolver: Resolver,
+    ):
+        self.filename = filename
+        self.current_version = current_version
+        self.current_version_build = current_version_build
+        self.base_version = base_version
+        self.base_version_build = base_version_build
```

Comment:
I would probably avoid duplicating all these data on each of the files, and pass the `FileTreeDiff` object to these files instead so they can grab these data from the parent: `self.diff.current_version`.